src/renderer/renderElement.ts

```diff
@@ -656,11 +666,12 @@ const generateElementWithCanvas = (
     prevElementWithCanvas &&
     prevElementWithCanvas.canvasZoom !== zoom.value &&
     !renderConfig?.shouldCacheIgnoreZoom;
-
+  const boundTextElementVersion = getBoundTextElement(element)?.version;
   if (
     !prevElementWithCanvas ||
     shouldRegenerateBecauseZoom ||
-    prevElementWithCanvas.theme !== renderConfig.theme
+    prevElementWithCanvas.theme !== renderConfig.theme ||
+    prevElementWithCanvas.boundTextElementVersion !== boundTextElementVersion
```

Comment:
`prevElementWithCanvas.boundTextElementVersion !== boundTextElementVersion` return true in most cases due to comparing `undefined` with `null`, making the canvas regenerate every time.